cocotb/handle.py

```diff
@@ -111,11 +111,11 @@ def __len__(self):
         return self._len
 
     def __eq__(self, other):
+        """Equality comparator for handles
 
-        # Permits comparison of handles i.e. if clk == dut.clk
-        if isinstance(other, SimHandleBase):
-            if self._handle == other._handle: return 0
-            return 1
+        IE if clk == dut.clk
+        """
+        return isinstance(other, SimHandleBase) and (self._handle == other._handle)
```

Comment:
This should be:
```python
if not isinstance(other, SimHandleBase):
    return NotImplemented
return self._handle == other._handle
```

Reply:
`False` or `NotImplemented` - if they have different base classes they're not equal. Is there a defined way of how Python likes to handle this?
If it is `Notimplemented`, do you use `return` or `raise`?

Reply:
`return NotImplemented` means "let the other class decide". If both classes return `NotImplemented`, then python will return false.